src/Microsoft.AspNetCore.Razor.Language/Extensions/DefaultTagHelperTargetExtension.cs

```diff
@@ -121,6 +128,12 @@ public void WriteTagHelperBody(CodeRenderingContext context, DefaultTagHelperBod
 
         public void WriteTagHelperCreate(CodeRenderingContext context, DefaultTagHelperCreateIntermediateNode node)
         {
+            if (context.Parent as TagHelperIntermediateNode == null)
+            {
+                var message = Resources.FormatIntermediateNodes_InvalidParentNode(node.GetType(), typeof(TagHelperIntermediateNode));
+                throw new InvalidOperationException(message);
```

Comment:
I decided to add some validation to these since we generally rely on `Context.Parent` being something we can look at.